.pre-commit-config.yaml

```diff
@@ -8,7 +8,18 @@ repos:
     -   id: end-of-file-fixer
     -   id: check-yaml
     -   id: check-added-large-files
--   repo: https://github.com/psf/black
-    rev: 24.2.0
+
+-   repo: local
     hooks:
     -   id: black
+        name: black
+        entry: poetry run black
+        language: system
+        types: [ python ]
+
+    -   id: poetry-lock
+        name: poetry lock
+        entry: poetry lock --no-update
+        language: system
+        types: [yaml]
+        files: ^pyproject\.toml$
```

Comment:
This hook keeps the `poetry.lock` file in sync with the dependencies specified in `pyproject.toml`.